backend/app/src/main/java/com/moh/phlat/backend/esb/boundary/PlrEsbBoundary.java

```diff
@@ -0,0 +1,209 @@
+
+	private PlrToken token;
+	
+	private SSLContext sslContext;
+
+	@Autowired
+    public void initSslContext(SslBundles sslBundles) throws NoSuchSslBundleException {
+		SslBundle sslBundle = sslBundles.getBundle("client");
+        sslContext = sslBundle.createSslContext();
+    }
+	
+	@Value("${plr.boundary.host}")
+	private String plrBoundaryHost;
+
+	@Value("${plr.boundary.keycloak.provider-url}")
+	private String keyCloakUrl;
+	@Value("${plr.boundary.keycloak.client-id}")
+	private String plrBoundaryClientId;
+	@Value("${plr.boundary.keycloak.client-secret}")
+	private String plrBoundaryClientSecret;
+	
+	private final String plrBoundaryEndpoint = "/HSA-web/rs/passthrough/maintain";
+	
+	@PostConstruct
+	public void initPlrEsbBoundary() {
+		try {
+			HttpClient httpClient = HttpClient.newBuilder()
+					// *TEMPORARY SSL DISABLER*
+					.sslContext(sslContext)
+					.build();
+			
+			webClient = WebClient.builder()
+					.clientConnector(new JdkClientHttpConnector(httpClient))
+					.build();
+			
+		} catch (Exception ex) {
+			logger.error("Could not build PlrEsbBoundary Connection: ", ex);
+		}
+	}
+	
+	public boolean isReadyToConnect() {
+		return webClient != null;
+	}
+	
+	public List<PlrResponse> loadPlrViaEsb(Control control, ProcessData processData) {
+		List<PlrResponse> plrResponses = new ArrayList();
+		token = getPlrKeyCloakDetails();
+		if (StringUtils.hasText(token.getAccessToken()) && !StringUtils.hasText(token.getError())) {
+			if (control.getLoadTypeFacility()) {
+				MaintainFacilityResponse facilityResponse = createFacility(control, processData);
+				plrResponses.add(facilityResponse);
+			}
+			if (control.getLoadTypeHds()) {
+				//MaintainHdsResponse hdsResponse = createHdsProvider(control, processData);
+				//plrResponses.add(hdsResponse);
+			}
+			if (control.getLoadTypeOFRelationship()) {
+				//MaintainOFResponse ofResponse = createOFRelationship(control, processData);
+				//plrResponses.add(ofResponse);
+			}
+		} 
+		return plrResponses;
+	}
+	
+	private MaintainFacilityResponse createFacility(Control control, ProcessData processData) {
+		MaintainFacilityRequest maintainFacilityRequest = new MaintainFacilityRequest(control);
+		String jsonRequest = maintainFacilityRequest.processDataToPlrJson(processData);
+		
+		MaintainFacilityResponse facilityResponse = new MaintainFacilityResponse(control);
+		
+		try {
+			Mono<String> mono = webClient.post()
+					.uri(plrBoundaryHost + plrBoundaryEndpoint)
+					.contentType(MediaType.APPLICATION_JSON)
+					.headers(h -> {
+						h.setBearerAuth(token.getAccessToken());
+						h.add("userID", control.getUserId());
+					})
+					.bodyValue(jsonRequest)
+					.retrieve()
+					.bodyToMono(String.class);
+			String jsonResponse = mono.block();
+			
+			facilityResponse.plrJsonToProcessData(jsonResponse, processData);
+			
+		} catch (Exception ex) {
+			logger.error("Load Facility request failed: ", ex);
+			facilityResponse.handlePlrError(ex);
```

Comment:
we need to discuss this. If we are running in loop, and there is system error, we need to do retry for a single record and abort the process if it is still not working as it indicates that PLR service is not available. Trying for all the records in file is not needed. What do you think?